UI/homepage.py

```diff
@@ -62,20 +142,55 @@ def draw_filter_search_screen():
     current_screen = 'filter_search'
     update_items_display()
 
+def draw_recipe_search_screen():
+    global current_screen, back_button, recipe_title, recipe_search_button, filter_buttons
+    
+    title.hide()
+    button_filter.hide()
+    button_recipe.hide()
+    back_button = pygame_gui.elements.UIButton(relative_rect=pygame.Rect(50, 550, 100, 40),
+                                               text='Back',
+                                               manager=manager)
+    recipe_title = pygame_gui.elements.UITextEntryLine(relative_rect=pygame.Rect(50, 100, 300, 50),
+                                                       manager=manager)
+    recipe_search_button = pygame_gui.elements.UIButton(relative_rect=pygame.Rect(200, 200, 200, 50),
+                                                        text='Recipe Search',
+                                                        manager=manager)
+    
+    for idx, feature in enumerate(['filter1       ', 'filter2', 'filter3']):
+        btn = pygame_gui.elements.UILabel(relative_rect=pygame.Rect(100, 300 + idx * 60, 200, 50),
+                                           text=feature,
+                                           manager=manager)
+        filter_buttons.append(btn)
+        
+        # Creating checkboxes
+        checkbox = Checkbox(screen, 150, 320 + idx * 60, idx, caption='', font_size=25)
+        check_boxes.append(checkbox)
+
+    current_screen = 'recipe_search'
+
 def update_items_display():
-    global delete_buttons
-    formatted_text = '<br>'.join(f'{item}' for item in item_list)
+    global delete_buttons, plus_button, minus_button, value_label, value
```

Comment:
similar to the last instance of creating global objects, you've already marked `plus_button` and `minus_button` as global back on line 122, if these are always the same object, then you should just initalize it outside of the functions and that will already set it as global by default, but here since they're called inside functions they're not going to be initalized until you run the function again so you'll probably have to go through these to check and see if they need to be initialized in a function, and if so which one because if you know for sure that by this point `plus_button` would have already been initialized then there's no need to reinitalize it

Reply:
it'll also be good to check for all of these variables to see if they're truly needed to be global or not